temporal-sdk/src/main/java/io/temporal/client/UpdateWithStartWorkflowOperation.java

```diff
@@ -0,0 +1,407 @@
+   * request.
+   *
+   * @param request method reference to a proxy created through {@link
+   *     WorkflowClient#newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first request function parameter
+   * @param arg2 second request function parameter
+   * @param arg3 third request function parameter
+   * @param arg4 fourth request function parameter
+   * @param arg5 sixth request function parameter
+   */
+  public static <A1, A2, A3, A4, A5> Builder<Void> newBuilder(
+      Functions.Proc5<A1, A2, A3, A4, A5> request, A1 arg1, A2 arg2, A3 arg3, A4 arg4, A5 arg5) {
+    return new Builder<>(() -> request.apply(arg1, arg2, arg3, arg4, arg5));
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation} for a six argument
+   * request.
+   *
+   * @param request method reference to a proxy created through {@link
+   *     WorkflowClient#newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first request function parameter
+   * @param arg2 second request function parameter
+   * @param arg3 third request function parameter
+   * @param arg4 fourth request function parameter
+   * @param arg5 sixth request function parameter
+   * @param arg6 sixth request function parameter
+   */
+  public static <A1, A2, A3, A4, A5, A6> Builder<Void> newBuilder(
+      Functions.Proc6<A1, A2, A3, A4, A5, A6> request,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5,
+      A6 arg6) {
+    return new Builder<>(() -> request.apply(arg1, arg2, arg3, arg4, arg5, arg6));
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation}.
+   *
+   * @param updateName name of the update handler
+   * @param resultClass class of the update return value
+   * @param args update request arguments
+   */
+  public static <R> Builder<R> newBuilder(String updateName, Class<R> resultClass, Object[] args) {
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setResultClass(resultClass).setUpdateName(updateName), args);
+  }
+
+  private WorkflowStub stub;
+
+  private UpdateOptions<R> options;
+
+  private Object[] updateArgs;
+
+  private final CompletableFuture<WorkflowUpdateHandle<R>> handle;
+
+  private final Functions.Proc request;
+
+  public UpdateWithStartWorkflowOperation(
+      UpdateOptions<R> options, Functions.Proc request, Object[] updateArgs) {
+    this.options = options;
+    this.updateArgs = updateArgs;
+    this.handle = new CompletableFuture<>();
+    this.request = request;
+  }
+
+  WorkflowUpdateHandle<R> invoke(Functions.Proc workflow) {
+    WorkflowInvocationHandler.initAsyncInvocation(
+        WorkflowInvocationHandler.InvocationType.UPDATE_WITH_START, this);
+    try {
+      request.apply();
+      workflow.apply();
+      stub.updateWithStart(this, this.updateArgs);
+      return this.handle.get();
+    } catch (ExecutionException | InterruptedException e) {
+      throw new RuntimeException(e);
```

Comment:
Is this the best way to convert this from a checked exception to an unchecked? (annotating all methods with the two unchecked exceptions seems undesirable to me)

Reply:
`ExecutionException ` should be unwrapped, it is just a wrapper around the real exception from the future. I think for `InterruptedException ` we should just set the interrupted flag so 
` Thread.currentThread().interrupt();`

Reply:
Ah, thank you for the pointer! I found another place where we do this now and will copy that approach (just like you described).

Reply:
The Java SDK does this in other places https://github.com/temporalio/sdk-java/blob/a173dbeb6e1b89f7ad184f2e3d9e21e185d51597/temporal-sdk/src/main/java/io/temporal/client/EagerPaginator.java#L99

Reply:
Make sure your test coverage also includes `getResult()` when the update has failed

Reply:
Added! https://github.com/temporalio/sdk-java/pull/2199/commits/c5d8b34d7c0b78a506e872f36816d2dff9e6c237